ambry-frontend/src/main/java/com/github/ambry/frontend/DeleteBlobHandler.java

```diff
@@ -160,6 +160,7 @@ private Callback<Void> routerCallback() {
         if (RestUtils.isDatasetVersionQueryEnabled(restRequest.getArgs())) {
           try {
             metrics.deleteDatasetVersionRate.mark();
+            accountAndContainerInjector.injectDatasetForNamedBlob(restRequest);
```

Comment:
Why do we have to set the dataset version here not in handle function?

Reply:
Because we need to make sure the dataset access is after the permission check which is done in processRequest in close source.